dynatrace/resource_aws_credentials_configuration.go

```diff
@@ -0,0 +1,136 @@
+	return &schema.Resource{
+		Create: resourceAwsCredentialsConfigutationCreate,
+		Read:   resourceAwsCredentialsConfigutationRead,
+		Update: resourceAwsCredentialsConfigutationUpdate,
+		Delete: resourceAwsCredentialsConfigutationtDelete,
+		Exists: resourceAwsCredentialsConfigutationExists,
+
+		Schema: map[string]*schema.Schema{
+			"label": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"partition_type": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"AWS_CN"}, false),
+			},
+			"authentication_data": {
+				Type: schema.TypeList,
+				Required:     true,
+				Elem: &schema.Resource {
+					Schema: map[string]*schema.Schema {
+						"type": &schema.Schema {
+							Type: schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{"KEYS"}, false),
+						},
+						"key_base_authentication": {
+							Type: schema.TypeList,
+							Optional:     true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema {
+									"access_key": &schema.Schema {
+										Type: schema.TypeString,
+										Required: true,
+									},
+									"secret_key": &schema.Schema {
+										Type: schema.TypeString,
+										Required: true,
+									},									
+								},
+							},
+						},
+						"role_base_authentication": {
+							Type: schema.TypeList,
+							Optional:     true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"iam_role": &schema.Schema {
+										Type: schema.TypeString,
+										Required: true,
+									},
+									"account_id": &schema.Schema {
+										Type: schema.TypeString,
+										Required: true,
+									},									
+								},
+							},
+						},
+					},
+				},
+			},
+			"tagged_only": &schema.Schema {
+				Type: schema.TypeBool,
+				Optional: true,
+				Default: true,
+			},
+			"tags_to_monitor": &schema.Schema {
+				Type: schema.TypeList,
+				Optional:     true,
+				Elem: &schema.Resource {
+					Schema: map[string]*schema.Schema {
+						"name": &schema.Schema {
+							Type: schema.TypeString,
+							Required: true,
+						},
+						"value": &schema.Schema {
+							Type: schema.TypeString,
+							Required: true,
+						},									
+					},
+				},
+			},
+			"supporting_services": &schema.Schema {
+				Type: schema.TypeList,
+				Optional:     true,
+				Elem: &schema.Resource {
+					Schema: map[string]*schema.Schema {
+						"name": &schema.Schema {
+							Type: schema.TypeString,
+							Required: true,
+						},
+						"metrics": &schema.Schema {
+							Type: schema.TypeList,
+							Required: true,
+							MinItems : 1,
+							Elem: &schema.Schema{
+							    Type: schema.TypeString,
+							},
+						},									
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsCredentialsConfigutationCreate(d *schema.ResourceData, meta interface{}) error {
+	return nil
```

Comment:
Create and update functions should always return the value of the read function. This helps keep terraform's internal state up to date.

From https://www.terraform.io/docs/extend/writing-custom-providers.html#defining-resources:

>Based on the schema and current state of the resource, Terraform can determine whether it needs to create a new resource, update an existing one, or destroy. The create and update function should always return the read function to ensure the state is reflected in the terraform.state file.

```
func resourceAwsCredentialsConfigutationCreate(d *schema.ResourceData, m interface{}) error {
        return resourceAwsCredentialsConfigutationRead(d, m)
}
```